### PR TITLE
[feature]: Store Switcher Groups

### PR DESCRIPTION
**Pull request by sirugh**
**! Only useable after 2.4.2 merge !**

## Description

Differentiate between stores in different groups.
Display "group name - view name" when multiple groups.
Display "view name" when only single group.

## Related Issue

Closes [PWA-1070](https://jira.corp.magento.com/browse/PWA-1070).

### Verification Steps

1. Use a 2.4.2 backend.
2. Set up multiple stores (groups)
3. Set up multiple views within each store
4. Visit ui and ensure switcher conforms to ux mocks
5. Ensure switcher has proper overflow when many groups/views are configured.
6. Ensure switcher has proper truncation when groups/views have long names.

## Screenshots / Screen Captures (if appropriate)
[Image from Gyazo](https://gyazo.com/8a4fc6f2ffe60e12c496f3bc5ac1f40b)

[Image from Gyazo](https://gyazo.com/c0b87e0d6cb8860c5ccded012a42f7d7)

Truncated long text in mobile: 

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: packages/venia-ui/lib/components/Header/currencySwitcher.css [hunk 1]
```diff
@@ -1,41 +1,102 @@
 .root {
-    composes: root from './storeSwitcher.css';
```
Comment (anthoula) [inline packages/venia-ui/lib/components/Header/currencySwitcher.css:2]: A few instances in this file where we've removed the `composes` properties, and replaced with duplicated styles. Are we avoiding `composes`? Any possibility of avoiding duplication?

Comment (sirugh) [reply packages/venia-ui/lib/components/Header/currencySwitcher.css:2]: The switchers are no longer identical with the introduction of store grouping. Styles that previously may have been generic/shareable are no longer. If anything, there should be some top-level `switcher.css` styles that both `storeSwitcher` and `currencySwitcher` compose from. Short of adding that scope to this PR I just duplicated the old styles to make sure nothing got lost.

**Review by anthoula — APPROVED** [review]

**Comment by dpatil-magento** [comment]
QA Approved.

File: packages/peregrine/lib/talons/Header/useStoreSwitcher.js [hunk 2]
```diff
@@ -19,18 +19,25 @@ const mapAvailableOptions = (config, stores) => {
...
             currency,
             isCurrent,
             locale,
             product_url_suffix,
             secure_base_media_url,
-            storeName
+            sortOrder,
+            storeName,
+            storeGroupName,
+            storeGroupCode
```
Comment (anthoula) [inline packages/peregrine/lib/talons/Header/useStoreSwitcher.js:40]: Alpha order is recommended

**Review by anthoula — APPROVED** [review]

**Comment by anthoula** [comment]
Looks great @sirugh  - thank you

**Pull request merged**